client/auth/edit-account.jsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+
+import Avatar from '../avatars/avatar.jsx'
+import Dialog from '../material/dialog.jsx'
+import FlatButton from '../material/flat-button.jsx'
+import form from '../forms/form.jsx'
+import LoadingIndicator from '../progress/dots.jsx'
+import PasswordTextField from '../material/password-text-field.jsx'
+import SubmitOnEnter from '../forms/submit-on-enter.jsx'
+
+import { closeDialog } from '../dialogs/action-creators'
+import { updateAccount } from '../auth/auther'
+import { composeValidators, matchesOther, minLength, required } from '../forms/validators'
+import { PASSWORD_MINLENGTH } from '../../common/constants'
+
+import { colorError } from '../styles/colors'
+import { Subheading } from '../styles/typography'
+
+const passwordValidator = composeValidators(
+  required('Enter a password'),
+  minLength(PASSWORD_MINLENGTH, `Enter at least ${PASSWORD_MINLENGTH} characters`),
+)
+const newPasswordValidator = composeValidators(
+  required('Enter a new password'),
+  minLength(PASSWORD_MINLENGTH, `Enter at least ${PASSWORD_MINLENGTH} characters`),
+)
+const confirmNewPasswordValidator = composeValidators(
+  required('Confirm your new password'),
+  matchesOther('newPassword', 'Enter a matching password'),
+)
+
+@form({
+  currentPassword: passwordValidator,
+  newPassword: newPasswordValidator,
+  confirmNewPassword: confirmNewPasswordValidator,
+})
+class AccountForm extends React.Component {
+  render() {
+    const { bindInput, onSubmit } = this.props
+    const textInputProps = {
+      autoCapitalize: 'off',
+      autoCorrect: 'off',
+      spellCheck: false,
+      tabIndex: 1,
+    }
+
+    return (
+      <form noValidate={true} onSubmit={onSubmit}>
+        <SubmitOnEnter />
+        <PasswordTextField
+          {...bindInput('currentPassword')}
+          label='Current password'
+          floatingLabel={true}
+          inputProps={textInputProps}
+        />
+        <PasswordTextField
```

Comment:
It doesn't really make sense atm, but will once we add email changing as well: I sort of like Discord's pattern here. Editing any of this information requires the current password, but the new password textboxes (new + confirm) aren't visible until users click a 'Change password?' link underneath the current password input. Would probably be good to do something similar here.

(And also, once email is added, the new password would only be required if you were trying to change it. It seems like Discord only validates if you've actually entered something into the new password box)

Reply:
Yeah, I've thought a bit about the UI/UX of this form once we allow email changes (and even profile image changes), but decided to leave it for now and implement it once we allow email changes. Which should be in the next PR :)

And yeah, I agree with your (and Discords) suggestion here. Although I have a couple of questions about that since we're on the subject:
- Should we display the current email in the text field that allows email changes?
- And if so, what's the best way to get the email of a user? Stick it in the session info that user gets initialized with?
- Should changing profile image require current password to be entered as well?

Reply:
Showing the current email seems fine, it's already available in the session and sent in the AUTH_UPDATE action, the store just doesn't do anything with it.

Changing avatar without entering the current password seems fine.